File: php/convert.php
<?php
$apiUrl = "https://api.sinhcoms.com/video/vr-2d/";
$videoUrl = isset($_GET['videoUrl']) ? $_GET['videoUrl'] : null;

if (!$videoUrl) {
    echo json_encode(['error' => 'Please provide a valid videoUrl.']);
    exit;
}
$fullUrl = $apiUrl . "?videoUrl=" . urlencode($videoUrl);
$curl = curl_init();
curl_setopt_array($curl, [
    CURLOPT_RETURNTRANSFER => true, 
    CURLOPT_URL => $fullUrl,        
    CURLOPT_HEADER => false,        
    CURLOPT_FOLLOWLOCATION => true
]);
$response = curl_exec($curl);
if ($response === false) {
    echo json_encode(['error' => curl_error($curl)]);
    exit;
}

curl_close($curl);
header('Content-Type: video/mp4');
echo $response;

File: python/convert.py
import requests
import sys

# Check if the video URL is provided?
if len(sys.argv) < 2:
    print("Usage: python convert.py <videoUrl>")
    sys.exit(1)
api_url = "https://api.sinhcoms.com/video/vr-2d/"
video_url = sys.argv[1]
response = requests.get(api_url, params={'videoUrl': video_url})
if response.status_code == 200:
    with open('output.mp4', 'wb') as f:
        f.write(response.content)
    print("Conversion complete: output.mp4")
else:
    print("Failed to convert the video. Error:", response.text)
